repose-aggregator/components/filters/header-normalization-filter/src/test/scala/org/openrepose/filters/headernormalization/HeaderNormalizationConfigurationSchemaTest.scala

```diff
@@ -0,0 +1,290 @@
+               |    </target>
+               |</header-normalization>""".stripMargin
+          val exception = intercept[SAXParseException] {
+            validator.validateConfigString(config)
+          }
+          exception.getLocalizedMessage should include("Invalid content")
+        }
+      }
+
+      it(s"should fail to validate when a target defines both a $filterType whitelist and blacklist") {
+        val config =
+          s"""<header-normalization xmlns='http://docs.openrepose.org/repose/header-normalization/v1.0'>
+             |    <target>
+             |        <$filterType>
+             |            <whitelist>
+             |                <header id="X-Some-Header"/>
+             |            </whitelist>
+             |            <blacklist>
+             |                <header id="X-Some-Header"/>
+             |            </blacklist>
+             |      </$filterType>
+             |    </target>
+             |</header-normalization>""".stripMargin
+        val exception = intercept[SAXParseException] {
+          validator.validateConfigString(config)
+        }
+        exception.getLocalizedMessage should include("Invalid content")
+      }
+    }
+  }
+
+  private val coreSpringProvider = CoreSpringProvider.getInstance()
+  coreSpringProvider.initializeCoreContext("/etc/repose", false)
+  private val reposeVersion = coreSpringProvider.getCoreContext.getEnvironment.getProperty(
+    ReposeSpringProperties.stripSpringValueStupidity(ReposeSpringProperties.CORE.REPOSE_VERSION))
+
+  if (reposeVersion != "10") {
+    describe("deprecated schema validation") {
+      it("should successfully validate when the deprecated header-filters is used") {
+        val config =
+          """<header-normalization xmlns='http://docs.openrepose.org/repose/header-normalization/v1.0'>
+            |    <header-filters>
+            |        <target>
+            |        </target>
+            |    </header-filters>
+            |</header-normalization>""".stripMargin
+        validator.validateConfigString(config)
+      }
+
+      it("should fail to validate when the deprecated header-filters is used with targets at the same level") {
+        val config =
+          """<header-normalization xmlns='http://docs.openrepose.org/repose/header-normalization/v1.0'>
+            |    <header-filters>
+            |        <target>
+            |        </target>
+            |    </header-filters>
+            |    <target>
+            |    </target>
+            |</header-normalization>""".stripMargin
+        val exception = intercept[SAXParseException] {
+          validator.validateConfigString(config)
+        }
+        exception.getLocalizedMessage should include("Cannot define the deprecated header-filters at this level along side a target")
+      }
+
+      listTypes.foreach { listType =>
+        it(s"should successfully validate when a deprecated header-filters' target defines two $listType") {
```

Comment:
This is a side effect of being able to define multiple whitelists and multiple blacklists on the same target. I have added TODO's that will make this fail, but currently you can define a whitelist and a blacklist on the same target.